api/provider/aws/service_delete.go

```diff
@@ -1,5 +1,131 @@
 package aws
 
-func (s *ServiceProvider) Delete(ServiceID string) error {
+import (
+	"github.com/aws/aws-sdk-go/service/ecs"
+)
+
+// Delete a specified service within a cluster.
+func (s *ServiceProvider) Delete(serviceID string) error {
+	desiredCount := 0
+
+	environmentID, err := lookupEntityEnvironmentID(s.TagStore, "service", serviceID)
+	if err != nil {
+		return err
+	}
+
+	clusterName := addLayer0Prefix(s.Config.Instance(), environmentID)
+	fqServiceID := addLayer0Prefix(s.Config.Instance(), serviceID)
+
+	if err := s.stopARNTasks(clusterName, fqServiceID); err != nil {
+		return err
+	}
+
+	if err := s.scaleService(clusterName, fqServiceID, desiredCount); err != nil {
+		return err
+	}
+
+	if err := s.deleteService(clusterName, fqServiceID); err != nil {
+		return err
+	}
+
+	if err := deleteEntityTags(s.TagStore, "service", serviceID); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *ServiceProvider) getARNTasks(clusterName, serviceID string,
+	deployments []*ecs.Deployment) ([]*string, error) {
+	taskARNs := []*string{}
+
+	for _, deployment := range deployments {
+		running := "RUNNING"
+		stopped := "STOPPED"
+
+		input := &ecs.ListTasksInput{}
+		input.SetCluster(clusterName)
+		input.SetDesiredStatus(running)
+		input.SetStartedBy(*deployment.Id)
+
+		tasks, err := s.AWS.ECS.ListTasks(input)
+		if err != nil {
+			return nil, err
+		}
+
+		taskARNs = append(taskARNs, tasks.TaskArns...)
+
+		input.SetDesiredStatus(stopped)
+		tasks, err = s.AWS.ECS.ListTasks(input)
+		if err != nil {
+			return nil, err
+		}
+
+		taskARNs = append(taskARNs, tasks.TaskArns...)
+	}
+	return taskARNs, nil
+}
+
+func (s *ServiceProvider) stopARNTasks(clusterName, serviceID string) error {
+	service, err := s.readService(clusterName, serviceID)
+	if err != nil {
+		return err
+	}
+
+	taskARNs, err := s.getARNTasks(clusterName, serviceID, service.Deployments)
```

Comment:
part of the pattern that we've been trying to adopt is to strongly avoid chaining helper functions. In other words, call all the helpers from the primary `Delete()` function and give the results of those functions clear variable names

Reply:
Got it, I'll change that.